sync.js

```diff
@@ -100,9 +100,15 @@ class SyncState {
   }
 
   stale (peer) {
+    // XXX: This is important because this peer can get in a state where it's in
+    // progress but the other side has not yet acknowlegded to us that it has
+    // finished downloading
+    // we can remove this check once we have ACKs or proper backpressure
+    var COMPLETE = peer.state.message.db.sofar !== peer.state.message.db.total &&
+      peer.state.message.media.sofar !== peer.state.message.media.total
     var staleDate = Date.now() - DEFAULT_HEARTBEAT_INTERVAL
     return peer.state.topic === ReplicationState.PROGRESS &&
-      (peer.state.message.timestamp < staleDate)
+      (peer.state.message.timestamp < staleDate) && COMPLETE
```

Comment:
```suggestion
      (peer.state.message.timestamp < staleDate) && INCOMPLETE
```
I was getting confused by this until I realized the variable name. ;)